R/connect.R

```diff
@@ -730,23 +712,13 @@ Connect <- R6::R6Class(
     #' @description Get remote groups.
     #' @param prefix The search term.
     #' @param limit The maximal result set size.
-    groups_remote = function(prefix = NULL, limit = 20) {
-      if (limit > 500) {
-        # reset limit to avoid error
-        limit <- 500
-      }
-      path <- glue::glue(
-        "v1/groups/remote?",
-        glue::glue(
-          safe_query(prefix, "prefix="),
-          safe_query(limit, "limit="),
-          .sep = "&"
-        ) %>%
-          gsub("^&+", "", .) %>%
-          gsub("&+", "&", .)
+    groups_remote = function(prefix = NULL, limit = 500) {
+      path <- v1_url("groups", "remote")
+      query <- list(
+        limit = valid_page_size(limit),
+        prefix = prefix
       )
-
-      self$GET(path)
+      self$GET(path, query = purrr::discard(query, is.null))
```

Comment:
This is fine as is since it's just a single function — but it does seem a little funny to have ever single query wrapped in this without some sort of helper (or we could override the `GET`, but that seems a little silly). 

But also, I thought `httr` did this stripping for us? But maybe some of [the setup and executing](https://github.com/rstudio/connectapi/pull/264/files#diff-5a4fc998698f639d47f38a272c1f3b10c3cd8316c72f0259037259dfbdd31369R141-R157)?

```
> query <- list(foo = NULL, bar = 1)
> httr::GET("https://jonkeane.com", query = query)
Response [https://jonkeane.com/?bar=1]
...
```

Anyway, not a huge must change, but it seems a little off

Reply:
I didn't even think to try, I'll strip these out and make sure tests pass.

Reply:
You're right, it just worked 🎉